### PR TITLE
[rpc][staking] Quick way to add an staking transaction error sink for…

### PR DESCRIPTION
**Pull request by fxfactorial**
Quick way to add a way for downstream consumers to see what is going wrong with latest staking transactions. 

```
loc2='http://localhost:9500'

 curl -d '{
    "jsonrpc": "2.0",
     "method": "hmy_getCurrentTransactionErrorSink",
     "params": [],
     "id": 1
}' -H "Content-Type: application/json" -X POST ${loc2}
```

### CODE REVIEW

File: internal/hmyapi/harmony.go [hunk 1]
```diff
@@ -65,3 +66,21 @@ func (s *PublicHarmonyAPI) GetNodeMetadata() NodeMetadata {
...
+func (s *PublicHarmonyAPI) GetCurrentTransactionErrorSink() []core.StakingTransactionError {
+	result := []core.StakingTransactionError{}
+	count := 0
+	core.StakingTransactionErrorSink.Range(func(key, value interface{}) bool {
+		count++
+		result = append(result, value.(core.StakingTransactionError))
+		return true
+	})
+	const maxMapSize = 1024
+	if count == maxMapSize {
```
Comment (rlan35) [inline internal/hmyapi/harmony.go]: This logic is confusing to me. Something is wrong.

Comment (fxfactorial) [reply internal/hmyapi/harmony.go]: If I reached the max size that I wanted to keep in memory, then delete it all

Comment (fxfactorial) [reply internal/hmyapi/harmony.go]: alternative i thought of using is a circular queue, or something circular data structure

File: node/rpc.go [hunk 2]
```diff
@@ -49,10 +49,17 @@ func (node *Node) IsCurrentlyLeader() bool {
 	return node.Consensus.IsLeader()
 }
 
+// ErroredStakingTransactionSink is the inmemory failed staking transactions this node has
+func (node *Node) ErroredStakingTransactionSink() []staking.RPCTransactionError {
+	return node.errorSink.failedTxns
```
Comment (rlan35) [inline node/rpc.go]: This will also need lock too. Reading when other writing.

**Pull request merged**